backend/tracim_backend/lib/utils/app.py

```diff
@@ -0,0 +1,110 @@
+from abc import ABC
+import typing
+
+from hapic.ext.pyramid import PyramidContext
+from pyramid.config import Configurator
+
+from tracim_backend.app_models.applications import TracimApplicationInContext
+from tracim_backend.models.roles import WorkspaceRoles
+
+if typing.TYPE_CHECKING:
+    from tracim_backend.config import CFG  # noqa:F401
+    from tracim_backend.app_models.contents import ContentStatus  # noqa:F401
+
+
+class TracimContentType(ABC):
+    def __init__(
+        self,
+        slug: str,
+        fa_icon: str,
+        label: str,
+        creation_label: str,
+        available_statuses: typing.List["ContentStatus"],
+        app: "TracimApplication" = None,
+        slug_alias: typing.List[str] = None,
+        allow_sub_content: bool = False,
+        file_extension: typing.Optional[str] = None,
+        minimal_role_content_creation: WorkspaceRoles = WorkspaceRoles.CONTRIBUTOR,
+    ) -> None:
+        self.slug = slug
+        self.fa_icon = fa_icon
+        self.label = label
+        self.creation_label = creation_label
+        self.available_statuses = available_statuses
+        self.slug_alias = slug_alias
+        self.allow_sub_content = allow_sub_content
+        self.file_extension = file_extension
+        self.minimal_role_content_creation = minimal_role_content_creation
+        self.app = app
+
+
+class TracimApplication(ABC):
+    def __init__(
+        self, label: str, slug: str, fa_icon: str, config: typing.Dict[str, str], main_route: str
```

Comment:
every string parameter should have an explanation about what values can be added. If I put "hey, bos ! what do you think about this?" is it pertinent for label / slug / fa_icon / config / main route ? give what type of string is expected, give examples

Reply:
no example for config as this parameter is not used now, but done for other